File: sdks/dotnet/NexusClient.cs
using System.Net.Http.Json;
using System.Text.Json;

namespace Nexus.SDK;

public class NexusError : Exception
{
    public int StatusCode { get; }
    public NexusError(int status, string message) : base($"HTTP {status}: {message}") => StatusCode = status;
}

public class NexusAuthError : NexusError
{
    public NexusAuthError(string message) : base(401, message) { }
}

public class NexusClient : IDisposable
{
    private readonly HttpClient _http;
    private readonly string _tenantId;
    private bool _disposed;

    public NexusClient(string baseUrl = "http://localhost:5000", string tenantId = "tenant-default")
    {
        _tenantId = tenantId;
        _http = new HttpClient { BaseAddress = new Uri(baseUrl.TrimEnd('/') + '/') };
        _http.DefaultRequestHeaders.Add("X-Tenant-ID", tenantId);
        _http.DefaultRequestHeaders.Add("Accept", "application/json");
    }

    public NexusClient(HttpClient httpClient, string tenantId = "tenant-default")
    {
        _http = httpClient;
        _tenantId = tenantId;
        _http.DefaultRequestHeaders.TryAddWithoutValidation("X-Tenant-ID", tenantId);
    }

    private async Task<T?> GetAsync<T>(string path, CancellationToken ct = default)
    {
        var resp = await _http.GetAsync(path, ct);
        await EnsureSuccess(resp, ct);
        return await resp.Content.ReadFromJsonAsync<T>(cancellationToken: ct);
    }

    private async Task<T?> PostAsync<T>(string path, object? body = null, CancellationToken ct = default)
    {
        HttpResponseMessage resp;
        if (body is null)
            resp = await _http.PostAsync(path, null, ct);
        else
            resp = await _http.PostAsJsonAsync(path, body, ct);
        await EnsureSuccess(resp, ct);
        return await resp.Content.ReadFromJsonAsync<T>(cancellationToken: ct);
    }

    private static async Task EnsureSuccess(HttpResponseMessage resp, CancellationToken ct)
    {
        if (resp.IsSuccessStatusCode) return;
        var body = await resp.Content.ReadAsStringAsync(ct);
        if ((int)resp.StatusCode == 401) throw new NexusAuthError(body);
        throw new NexusError((int)resp.StatusCode, body);
    }

    // ── Agents ──────────────────────────────────────────────────────────────
    public Task<JsonElement?> ListAgentsAsync(int page = 1, int pageSize = 50, CancellationToken ct = default)
        => GetAsync<JsonElement>($"api/agents?page={page}&pageSize={pageSize}", ct);

    public Task<JsonElement?> GetAgentAsync(string agentId, CancellationToken ct = default)
        => GetAsync<JsonElement>($"api/agents/{Uri.EscapeDataString(agentId)}", ct);

    public Task<JsonElement?> CreateAgentAsync(string name, string capability, string? swarmId = null, CancellationToken ct = default)
        => PostAsync<JsonElement>("api/agents", new { name, capability, tenantId = _tenantId, swarmId }, ct);

    public Task<JsonElement?> ExecuteAgentTaskAsync(string agentId, string taskType, object? parameters = null, CancellationToken ct = default)
        => PostAsync<JsonElement>($"api/agents/{Uri.EscapeDataString(agentId)}/execute", new { taskType, parameters }, ct);

    public Task<JsonElement?> GetAgentStatsAsync(CancellationToken ct = default)
        => GetAsync<JsonElement>("api/agents/stats", ct);

    // ── Swarms ───────────────────────────────────────────────────────────────
    public Task<JsonElement?> ListSwarmsAsync(CancellationToken ct = default)
        => GetAsync<JsonElement>("api/swarms", ct);

    public Task<JsonElement?> GetSwarmAsync(string swarmId, CancellationToken ct = default)
        => GetAsync<JsonElement>($"api/swarms/{Uri.EscapeDataString(swarmId)}", ct);

    public Task<JsonElement?> RunConsensusAsync(string swarmId, string proposalId, string proposal, CancellationToken ct = default)
        => PostAsync<JsonElement>($"api/swarms/{Uri.EscapeDataString(swarmId)}/consensus", new { proposalId, proposal }, ct);

    // ── Crypto ───────────────────────────────────────────────────────────────
    public Task<JsonElement?> KyberKeypairAsync(string level = "Kyber768", CancellationToken ct = default)
        => PostAsync<JsonElement>($"api/crypto/kyber/keypair?level={level}", null, ct);

    public Task<JsonElement?> DilithiumKeypairAsync(string level = "Dilithium3", CancellationToken ct = default)
        => PostAsync<JsonElement>($"api/crypto/dilithium/keypair?level={level}", null, ct);

    public Task<JsonElement?> ShamirSplitAsync(string secretB64, int threshold, int total, CancellationToken ct = default)
        => PostAsync<JsonElement>("api/crypto/shamir/split", new { secret = secretB64, threshold, total }, ct);

    // ── Health ───────────────────────────────────────────────────────────────
    public Task<JsonElement?> HealthAsync(CancellationToken ct = default)
        => GetAsync<JsonElement>("health", ct);

    public void Dispose()
    {
        if (!_disposed) { _http.Dispose(); _disposed = true; }
    }
}

File: sdks/java/src/main/java/com/nexus/sdk/NexusClient.java
package com.nexus.sdk;

import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.net.URLEncoder;
import java.nio.charset.StandardCharsets;

/**
 * Nexus HyperIntelligence Java SDK Client.
 * Zero external HTTP dependency — uses java.net.HttpURLConnection.
 */
public class NexusClient {

    private final String baseUrl;
    private final String tenantId;
    private final int timeoutMs;
    private final ObjectMapper mapper = new ObjectMapper();

    public NexusClient(String baseUrl, String tenantId) {
        this.baseUrl = baseUrl.replaceAll("/$", "");
        this.tenantId = tenantId;
        this.timeoutMs = 30_000;
    }

    public NexusClient() {
        this("http://localhost:5000", "tenant-default");
    }

    // ── Internal ─────────────────────────────────────────────────────────────

    private JsonNode request(String method, String path, Object body) throws IOException, NexusException {
        URL url = new URL(baseUrl + path);
        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
        conn.setRequestMethod(method);
        conn.setRequestProperty("Content-Type", "application/json");
        conn.setRequestProperty("Accept", "application/json");
        conn.setRequestProperty("X-Tenant-ID", tenantId);
        conn.setConnectTimeout(timeoutMs);
        conn.setReadTimeout(timeoutMs);

        if (body != null) {
            conn.setDoOutput(true);
            byte[] data = mapper.writeValueAsBytes(body);
            try (OutputStream os = conn.getOutputStream()) {
                os.write(data);
            }
        }

        int status = conn.getResponseCode();
        InputStream stream = (status < 400) ? conn.getInputStream() : conn.getErrorStream();
        byte[] bytes = (stream != null) ? stream.readAllBytes() : new byte[0];

        if (status < 200 || status >= 300) {
            throw new NexusException(status, new String(bytes, StandardCharsets.UTF_8));
        }
        return bytes.length > 0 ? mapper.readTree(bytes) : mapper.createObjectNode();
    }

    // ── Agents ───────────────────────────────────────────────────────────────

    public JsonNode listAgents(int page, int pageSize) throws IOException, NexusException {
        return request("GET", "/api/agents?page=" + page + "&pageSize=" + pageSize, null);
    }

    public JsonNode getAgent(String agentId) throws IOException, NexusException {
        return request("GET", "/api/agents/" + URLEncoder.encode(agentId, StandardCharsets.UTF_8), null);
    }

    public JsonNode createAgent(String name, String capability) throws IOException, NexusException {
        ObjectNode body = mapper.createObjectNode()
            .put("name", name)
            .put("capability", capability)
            .put("tenantId", tenantId);
        return request("POST", "/api/agents", body);
    }

    // ── Swarms ────────────────────────────────────────────────────────────────

    public JsonNode listSwarms() throws IOException, NexusException {
        return request("GET", "/api/swarms", null);
    }

    public JsonNode runConsensus(String swarmId, String proposalId, String proposal) throws IOException, NexusException {
        ObjectNode body = mapper.createObjectNode()
            .put("proposalId", proposalId)
            .put("proposal", proposal);
        return request("POST", "/api/swarms/" + URLEncoder.encode(swarmId, StandardCharsets.UTF_8) + "/consensus", body);
    }

    // ── Crypto ────────────────────────────────────────────────────────────────

    public JsonNode kyberKeypair(String level) throws IOException, NexusException {
        if (level == null || level.isBlank()) level = "Kyber768";
        return request("POST", "/api/crypto/kyber/keypair?level=" + level, null);
    }

    public JsonNode shamirSplit(String secretB64, int threshold, int total) throws IOException, NexusException {
        ObjectNode body = mapper.createObjectNode()
            .put("secret", secretB64)
            .put("threshold", threshold)
            .put("total", total);
        return request("POST", "/api/crypto/shamir/split", body);
    }

    // ── Health ────────────────────────────────────────────────────────────────

    public JsonNode health() throws IOException, NexusException {
        return request("GET", "/health", null);
    }
}
